iterator/git.go

```diff
@@ -203,11 +209,14 @@ func (obj *Git) validateRef() error {
 	if obj.Ref == "" {
 		return fmt.Errorf("empty ref")
 	}
+	if obj.Ref != "" && strings.Contains(obj.Ref, separator) {
```

Comment:
let's consider creating another validateXXX sub method...

Reply:
Fixed, Made a validateRev and validateHash.